### PR TITLE
[CPM-117] refactor common components

### PR DESCRIPTION
**Pull request by HongShi10**
refactored inline editing and table components

### CODE REVIEW

**Review by Dinith1 — APPROVED** [review]
Noice LGTM :shipit:

**Review by vee-christella — CHANGES_REQUESTED** [review]
Before you merge this, can you please have a look at the AlertDialog with Courses and Students and see if you can use `ConfirmationDialog` in common?

File: client/src/components/InlineEdit.js [hunk 1]
```diff
@@ -1,20 +1,21 @@
...
 
-const InlineEdit = ({ onSubmit, onCancel, onChange, value, title, isDisabled }) => {
+const InlineEdit = ({ onSubmit, onCancel, onChange, value, title, placeholder }) => {
     return (
         <Editable
             value={value}
             onSubmit={onSubmit}
+            onCancel={onCancel}
             onChange={onChange}
-            isDisabled={isDisabled}
```
[the remark this answers is not in the document]
Comment (Dinith1) [reply client/src/components/InlineEdit.js:10]: Yup we never use it

**Review by Richardng12 — APPROVED** [review]
lgtm fixed vanessas review

**Review by Dinith1 — APPROVED** [review]

File: client/src/components/index.js [hunk 2]
```diff
@@ -2,7 +2,10 @@ import AppConfiguration from "./AppConfiguration.js";
 import OutlineButton from "./OutlineButton.js";
 import ProgrammeCard from "./ProgrammeCard.js";
 import SearchBar from "./SearchBar.js";
-import ConfirmationDialog from "./ConfirmationDialog.js";
+// import ConfirmationDialog from "./ConfirmationDialog.js";
```
Comment (vee-christella) [inline client/src/components/index.js:5]: Please delete this

**Review by vee-christella — APPROVED** [review]
LGTM 🦊

**Pull request merged**